homeassistant/components/zha/fan.py

```diff
@@ -164,30 +171,60 @@ def __init__(
         self._available: bool = False
         group = self.zha_device.gateway.get_group(self._group_id)
         self._fan_channel = group.endpoint[hvac.Fan.cluster_id]
+        self._percentage = None
+        self._preset_mode = None
 
-        # what should we do with this hack?
-        async def async_set_speed(value) -> None:
-            """Set the speed of the fan."""
-            try:
-                await self._fan_channel.write_attributes({"fan_mode": value})
-            except ZigbeeException as ex:
-                self.error("Could not set speed: %s", ex)
-                return
+    @property
+    def percentage(self) -> str:
```

Comment:
```py
def percentage(self) -> Optional[int]:

Reply:
Fixed in https://github.com/home-assistant/core/pull/46714